### PR TITLE
Fix for inter-schema foreign keys

### PR DESCRIPTION
**Pull request by SamuelSchlesinger**
This is a fix for #222. Tested it locally and it works.

### CODE REVIEW

File: squeal-postgresql/src/Squeal/PostgreSQL/Type/Schema.hs [hunk 1]
```diff
@@ -210,7 +210,7 @@ data TableConstraint
   = Check [Symbol]
   | Unique [Symbol]
   | PrimaryKey [Symbol]
-  | ForeignKey [Symbol] Symbol [Symbol]
+  | ForeignKey [Symbol] (Symbol, Symbol) [Symbol]
```
Comment (echatav) [inline squeal-postgresql/src/Squeal/PostgreSQL/Type/Schema.hs:213]: might be cleaner without the pair?

File: squeal-postgresql/src/Squeal/PostgreSQL/Definition/Constraint.hs [hunk 2]
```diff
@@ -286,16 +289,17 @@ foreignKey keys parent refs ondel onupd = UnsafeTableConstraintExpression $
 
 -- | A constraint synonym between types involved in a foreign key constraint.
 type ForeignKeyed db
-  sch
-  schema
+  sch0 sch1
+  schema0 schema1
```
Comment (echatav) [inline squeal-postgresql/src/Squeal/PostgreSQL/Definition/Constraint.hs:293]: this was my mistake but the variable order goes parent-child here and then child-parent below

File: squeal-postgresql/src/Squeal/PostgreSQL/Type/Schema.hs (hunk 1 above)
Comment (SamuelSchlesinger) [reply squeal-postgresql/src/Squeal/PostgreSQL/Type/Schema.hs:213]: I think it makes sense to keep the pair and create some sugar like `"sch0" ! "tab"`

File: squeal-postgresql/src/Squeal/PostgreSQL/Definition/Constraint.hs (hunk 2 above)
Comment (SamuelSchlesinger) [reply squeal-postgresql/src/Squeal/PostgreSQL/Definition/Constraint.hs:293]: Yeah, my bad for blindly copying it over. Will make sure I understand it next time I push.

File: squeal-postgresql/src/Squeal/PostgreSQL/Type/Schema.hs (hunk 1 above)
Comment (echatav) [reply squeal-postgresql/src/Squeal/PostgreSQL/Type/Schema.hs:213]: I already have `:::` and `:=>` for type level pairs, trying not to add _too_ many new type operators

Comment (SamuelSchlesinger) [reply squeal-postgresql/src/Squeal/PostgreSQL/Type/Schema.hs:213]: Right, well I'll leave the choice up to you, but I think its more intuitive to make it as similar to the `#sch ! #tab` syntax as possible.

**Comment by echatav** [comment]
We broke doctests :-)

**Comment by SamuelSchlesinger** [comment]
I think I fixed them! Hopefully I didn't miss anything.

**Comment by SamuelSchlesinger** [comment]
Note for posterity: this commit has been recrafted to remove the swp file.

**Pull request merged**